eng/common/pipelines/templates/stages/archetype-typespec-emitter.yml

```diff
@@ -0,0 +1,452 @@
+# Produces the artifact `publish_artifacts` which contains the following:
+#   emitter-package.json: Created using the package json from the build step. 
+#   emitter-package-lock.json: Created by calling `npm install` using `emitter-package.json`
+- ${{ if parameters.ShouldPublish }}:
+  - stage: Publish
+    dependsOn:
+    - Build
+    - ${{ if and(parameters.PublishDependsOnTest, ne(length(parameters.TestMatrix), 0)) }}:
+      - Test
+    variables:
+      buildArtifactsPath: $(Pipeline.Workspace)/build_artifacts
+    pool: ${{ parameters.Pool }}
+    jobs:
+    - job: Publish
+      steps:
+      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+
+      - download: current
+        artifact: build_artifacts
+        displayName: Download build artifacts
+
+      # Create authenticated .npmrc file for publishing
+      - ${{ if eq(parameters.PublishInternal, 'true') }}:
+        - template: /eng/common/pipelines/templates/steps/create-authenticated-npmrc.yml
+          parameters:
+            npmrcPath: $(buildArtifactsPath)/packages/.npmrc
+            registryUrl: https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-js-test-autorest/npm/registry/
+      - ${{ else }}:
+        - pwsh: |
+            "//registry.npmjs.org/:_authToken=$(azure-sdk-npm-token)" | Out-File '.npmrc'
+          displayName: Authenticate .npmrc for npmjs.org
+          workingDirectory: $(buildArtifactsPath)/packages
+
+      # per package, publishing using appropriate tool
+      - ${{ each package in parameters.Packages }}:
+        - ${{ if eq(package.type, 'npm') }}:
+          - pwsh: |
+              $file = Resolve-Path "${{ package.file }}"
+              Write-Host "npm publish $file --verbose --access public --prefix $(buildArtifactsPath)/packages"
+              npm publish $file --verbose --access public --prefix $(buildArtifactsPath)/packages
+            displayName: Publish ${{ package.name }}
+            workingDirectory: $(buildArtifactsPath)/packages
+        - ${{ elseif eq(package.type, 'nuget') }}:
+          - task: NuGetCommand@2
+            displayName: Publish ${{ package.name }}
+            inputs:
+              command: 'push'
+              packagesToPush: $(buildArtifactsPath)/packages/${{ package.file }}
+              # Nuget packages are always published to the same internal feed https://dev.azure.com/azure-sdk/public/_packaging?_a=feed&feed=azure-sdk-for-net
+              nuGetFeedType: 'internal'
+              publishVstsFeed: '29ec6040-b234-4e31-b139-33dc4287b756/fa8c16a3-dbe0-4de2-a297-03065ec1ba3f'
+
+        - task: PowerShell@2
+          displayName: Create emitter-package.json
+          inputs:
+            pwsh: true
+            filePath: ./eng/common/scripts/typespec/New-EmitterPackageJson.ps1
+            arguments: >
+              -PackageJsonPath '$(buildArtifactsPath)/lock-files/package.json'
```

Comment:
This might need to be a parameter to the template if we end up needing to have multiple

Reply:
I believe we'll be convention based if we go to multiple. e.g. `emitter-package-{packageName}.json`
\If that's true, I'd want to generate the path from the package rather than have it be configured